crates/bevy_render/src/mesh/mesh.rs

```diff
@@ -502,16 +515,55 @@ pub fn mesh_resource_provider_system(
     mesh_events: Res<Events<AssetEvent<Mesh>>>,
     mut query: Query<(&Handle<Mesh>, &mut RenderPipelines)>,
 ) {
-    let vertex_buffer_descriptor = match state.vertex_buffer_descriptor {
-        Some(value) => value,
-        None => {
-            // TODO: allow pipelines to specialize on vertex_buffer_descriptor and index_format
-            let vertex_buffer_descriptor = Vertex::as_vertex_buffer_descriptor();
-            vertex_buffer_descriptors.set(vertex_buffer_descriptor.clone());
-            state.vertex_buffer_descriptor = Some(vertex_buffer_descriptor);
-            vertex_buffer_descriptor
-        }
-    };
+    if state.vertex_buffer_descriptor.is_none() {
+        // TODO: allow pipelines to specialize on vertex_buffer_descriptor and index_format
+        let mut vertex_buffer_descriptors_: VertexBufferDescriptors = VertexBufferDescriptors{
+            descriptors: Default::default()
+        };
+
+        // TODO: kinda messy
+        vertex_buffer_descriptors_.descriptors.insert(
+            "Vertex_Position".to_string(),
+            VertexBufferDescriptor{
+                name: "Vertex_Position".into(),
+                stride: VertexFormat::Float3.get_size(),
+                step_mode: InputStepMode::Vertex,
+                attributes: vec![VertexAttributeDescriptor{
+                    name: "Vertex_Position".into(),
+                    offset: 0,
+                    format: VertexFormat::Float3,
+                    shader_location: 0
+                }]
+        });
+        vertex_buffer_descriptors_.descriptors.insert(
+            "Vertex_Normal".to_string(),
+            VertexBufferDescriptor{
+                name: "Vertex_Normal".into(),
+                stride: VertexFormat::Float3.get_size(),
+                step_mode: InputStepMode::Vertex,
+                attributes: vec![VertexAttributeDescriptor{
+                    name: "Vertex_Normal".into(),
+                    offset: 0,
+                    format: VertexFormat::Float3,
+                    shader_location: 0
+                }]
+        });
+        vertex_buffer_descriptors_.descriptors.insert(
+            "Vertex_Uv".to_string(),
+            VertexBufferDescriptor{
+                name: "Vertex_Uv".into(),
+                stride: VertexFormat::Float2.get_size(),
+                step_mode: InputStepMode::Vertex,
+                attributes: vec![VertexAttributeDescriptor{
+                    name: "Vertex_Uv".into(),
+                    offset: 0,
+                    format: VertexFormat::Float2,
+                    shader_location: 0
+                }]
+        });
+        vertex_buffer_descriptors.set_many(vertex_buffer_descriptors_);
+    }
```

Comment:
This whole thing can be scrapped now I think. It only got used when doing mesh.get_vertex_buffer_bytes() which now never gets called anyway:

This system shouldn't care about a specific vertex layout, it should simply upload the data that exists in the mesh to the GPU.

Reply:
Yeah, however this is still sorta used to "confirm" the vertex layout in https://github.com/bevyengine/bevy/blob/408114269b36cbc088dc9b3a0a58ba8acba21acf/crates/bevy_render/src/pipeline/pipeline_layout.rs#L71-L91

If not set, bevy will panic with "Encountered unsupported Vertex Buffer". I strongly agree on that we should simplify this however.